Problems on the tests related to status code and others

Hi,

Some of the end-to-end tests are failing due to wrong status code, incorrect use of json.dumps as parameter to POST requests library and other minor errors. Examples:

```
>       assert response.status_code == 200
E       AssertionError: assert 201 == 200
E        +  where 201 = <Response [201]>.status_code

tests/test_e2e_10_mef_eline.py:417: AssertionError
>       assert response.status_code == 200
E       AssertionError: assert 201 == 200
E        +  where 201 = <Response [201]>.status_code

tests/test_e2e_10_mef_eline.py:265: AssertionError
        response = requests.post(api_url, json=json.dumps(payload))
>       self.assertEqual(response.status_code, 200)
E       AssertionError: 500 != 200

tests/test_e2e_10_mef_eline.py:55: AssertionError
```

--- a/tests/test_e2e_05_topology.py
+++ b/tests/test_e2e_05_topology.py
@@ -67,10 +67,10 @@ class TestE2ETopology(unittest.TestCase):
         self.net.start_controller(clean_config=False)
         self.net.wait_switches_connect()
 
-        # restore the status
-        api_url = KYTOS_API+'/topology/v3/restore'
-        response = requests.get(api_url)
-        self.assertEqual(response.status_code, 200)
+        ## restore the status
+        #api_url = KYTOS_API+'/topology/v3/restore'
+        #response = requests.get(api_url)
+        #self.assertEqual(response.status_code, 200)
 
         # check if the switches are still enabled and now with the links
         api_url = KYTOS_API+'/topology/v3/switches'
@@ -118,10 +118,10 @@ class TestE2ETopology(unittest.TestCase):
         self.net.start_controller(clean_config=False)
         self.net.wait_switches_connect()
 
-        # restore the status
-        api_url = KYTOS_API+'/topology/v3/restore'
-        response = requests.get(api_url)
-        self.assertEqual(response.status_code, 200)
+        ## restore the status
+        #api_url = KYTOS_API+'/topology/v3/restore'
+        #response = requests.get(api_url)
+        #self.assertEqual(response.status_code, 200)
 
         # check if the interfaces are still enabled and now with the links
         api_url = KYTOS_API+'/topology/v3/interfaces'
@@ -191,10 +191,10 @@ class TestE2ETopology(unittest.TestCase):
         self.net.start_controller(clean_config=False)
         self.net.wait_switches_connect()
 
-        # restore the status
-        api_url = KYTOS_API+'/topology/v3/restore'
-        response = requests.get(api_url)
-        self.assertEqual(response.status_code, 200)
+        ## restore the status
+        #api_url = KYTOS_API+'/topology/v3/restore'
+        #response = requests.get(api_url)
+        #self.assertEqual(response.status_code, 200)
 
         # wait 10s to kytos execute LLDP
         time.sleep(10)

--- a/tests/test_e2e_05_topology.py
+++ b/tests/test_e2e_05_topology.py
@@ -159,14 +159,7 @@ class TestE2ETopology(unittest.TestCase):
         response = requests.get(api_url)
         self.assertEqual(response.status_code, 200)
         data = response.json()
-        self.assertEqual(len(data['links']), 3)
-        link_id1 = None
-        for k,v in data['links'].items():
-            link_a, link_b = v['endpoint_a']['id'], v['endpoint_b']['id']
-            if set([link_a, link_b]) == set([endpoint_a, endpoint_b]):
-                link_id1 = k
-        self.assertNotEqual(link_id1, None)
-        self.assertFalse(data['links'][link_id1]['enabled'])
+        self.assertEqual(len(data['links']), 0)
 
         # enable the links (need to enable the switches and ports first)
         for i in [1,2,3]:
@@ -177,6 +170,24 @@ class TestE2ETopology(unittest.TestCase):
             api_url = KYTOS_API+'/topology/v3/interfaces/switch/%s/enable' % (sw)
             response = requests.post(api_url)
             self.assertEqual(response.status_code, 200)
+
+        # wait 10s to kytos execute LLDP
+        time.sleep(10)
+
+        # now all the links should stay disabled
+        api_url = KYTOS_API+'/topology/v3/links'
+        response = requests.get(api_url)
+        data = response.json()
+        self.assertEqual(len(data['links']), 3)
+
+        link_id1 = None
+        for k,v in data['links'].items():
+            link_a, link_b = v['endpoint_a']['id'], v['endpoint_b']['id']
+            if set([link_a, link_b]) == set([endpoint_a, endpoint_b]):
+                link_id1 = k
+        self.assertNotEqual(link_id1, None)
+        self.assertFalse(data['links'][link_id1]['enabled'])
+
         api_url = KYTOS_API+'/topology/v3/links/%s/enable' % (link_id1)
         response = requests.post(api_url)
         self.assertEqual(response.status_code, 201)

--- a/tests/test_e2e_10_mef_eline.py
+++ b/tests/test_e2e_10_mef_eline.py
@@ -90,7 +90,7 @@ class TestE2EMefEline(unittest.TestCase):
         }
         api_url = KYTOS_API + '/mef_eline/v2/evc/'
         response = requests.post(api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})
-        assert response.status_code == 200
+        assert response.status_code == 201
         data = response.json()
         assert 'circuit_id' in data
         time.sleep(5)
@@ -140,7 +140,7 @@ class TestE2EMefEline(unittest.TestCase):
         }
         api_url = KYTOS_API + '/mef_eline/v2/evc/'
         response = requests.post(api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})
-        assert response.status_code == 200
+        assert response.status_code == 201
         data = response.json()
         assert 'circuit_id' in data
         time.sleep(5)
@@ -189,7 +189,7 @@ class TestE2EMefEline(unittest.TestCase):
         }
         api_url = KYTOS_API + '/mef_eline/v2/evc/'
         response = requests.post(api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})
-        assert response.status_code == 200
+        assert response.status_code == 201
         data = response.json()
         assert 'circuit_id' in data
         time.sleep(5)
@@ -240,7 +240,7 @@ class TestE2EMefEline(unittest.TestCase):
         }
         api_url = KYTOS_API + '/mef_eline/v2/evc/'
         response = requests.post(api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})
-        assert response.status_code == 200
+        assert response.status_code == 201
         data = response.json()
         assert 'circuit_id' in data
         evc1 = data['circuit_id']
@@ -262,7 +262,7 @@ class TestE2EMefEline(unittest.TestCase):
         }
         api_url = KYTOS_API + '/mef_eline/v2/evc/'
         response = requests.post(api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})
-        assert response.status_code == 200
+        assert response.status_code == 201
         data = response.json()
         assert 'circuit_id' in data
         evc2 = data['circuit_id']
@@ -385,7 +385,7 @@ class TestE2EMefEline(unittest.TestCase):
         }
         api_url = KYTOS_API + '/mef_eline/v2/evc/'
         response = requests.post(api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})
-        assert response.status_code == 200
+        assert response.status_code == 201
         data = response.json()
         assert 'circuit_id' in data
         evc1 = data['circuit_id']
@@ -414,7 +414,7 @@ class TestE2EMefEline(unittest.TestCase):
         }
         api_url = KYTOS_API + '/mef_eline/v2/evc/'
         response = requests.post(api_url, data=json.dumps(payload), headers={'Content-type': 'application/json'})
-        assert response.status_code == 200
+        assert response.status_code == 201
         data = response.json()
         assert 'circuit_id' in data
         evc2 = data['circuit_id']

--- a/tests/test_e2e_10_mef_eline.py
+++ b/tests/test_e2e_10_mef_eline.py
@@ -43,7 +43,7 @@ class TestE2EMefEline(unittest.TestCase):
                 }
             },
             "uni_z": {
-                "interface_id": "00:00:00:00:00:00:00:01:4",
+                "interface_id": "00:00:00:00:00:00:00:01:2",
                 "tag": {
                     "tag_type": 1,
                     "value": 101
@@ -51,22 +51,34 @@ class TestE2EMefEline(unittest.TestCase):
             }
         }
         api_url = KYTOS_API + '/mef_eline/v2/evc/'
-        response = requests.post(api_url, json=json.dumps(payload))
-        self.assertEqual(response.status_code, 200)
-
-        h1, h2 = self.net.get('h1', 'h2')
-        result = h1.cmd('ping -c1', h2.IP())
-        self.assertIn(', 0% packet loss,', result)
+        response = requests.post(api_url, json=payload)
+        self.assertEqual(response.status_code, 201)
         data = response.json()
         self.assertIn('circuit_id', data)
 
-        s1 = self.net.get('s1')
+        h11, h12 = self.net.net.get('h11', 'h12')
+        h11.cmd('ip link add link %s name vlan101 type vlan id 101' % (h11.intfNames()[0]))
+        h11.cmd('ip link set up vlan101')
+        h11.cmd('ip addr add 10.1.1.11/24 dev vlan101')
+        h12.cmd('ip link add link %s name vlan101 type vlan id 101' % (h12.intfNames()[0]))
+        h12.cmd('ip link set up vlan101')
+        h12.cmd('ip addr add 10.1.1.12/24 dev vlan101')
+
+        result = h11.cmd('ping -c1 10.1.1.12')
+        self.assertIn(', 0% packet loss,', result)
+
+        s1 = self.net.net.get('s1')
         flows_s1 = s1.dpctl('dump-flows')
         # Each switch must have 3 flows: 01 for LLDP + 02 for the EVC (ingress + egress)
         self.assertEqual(len(flows_s1.split('\r\n ')), 3)
 
         # TODO: make sure it should be dl_vlan instead of vlan_vid
-        self.assertIn('dl_vlan=100', flows_s1)
+        self.assertIn('dl_vlan=101', flows_s1)
+
+        # clean up
+        h11.cmd('ip link del vlan101')
+        h12.cmd('ip link del vlan101')
+        self.net.restart_kytos_clean()
 
     def test_015_create_evc_inter_switch(self):
         payload = {

